src/scenes/Partnerships/PartnershipForm/renderFields.tsx

```diff
@@ -0,0 +1,118 @@
+import { FormApi } from 'final-form';
+import React, { useEffect, useRef } from 'react';
+import {
+  hasManagingType,
+  PartnershipFormFragment,
+  PartnershipformValues,
+} from '.';
+import {
+  displayFinancialReportingType,
+  displayPartnershipStatus,
+  PartnershipAgreementStatus,
+  PartnershipAgreementStatusList,
+  PartnerType,
+} from '../../../api';
+import {
+  EnumField,
+  EnumFieldProps,
+  SecuredField,
+  SubmitError,
+} from '../../../components/form';
+import { PartnerField } from '../../../components/form/Lookup';
+import { Nullable } from '../../../util';
+
+export const renderFields = (partnership?: PartnershipFormFragment) => ({
```

Comment:
This is a React component, so it should be named `RenderFields`, not `renderFields`

Reply:
Well technically it's a function that takes in a partnership object and returns a react component, I named it this way to make the distinction and make sure the next developer knows it's different.  I'm not sure if this thing has a formal name or what the best naming convention is.  Thoughts?

Reply:
Yeah, I guess you're right... my eyes kind of glossed over the fact that you weren't passing props but a single argument. But you're calling `useEffect`, so my thinking is this *should* be a React component and should be named like one and called like one.

Reply:
(Because otherwise what you're technically doing is calling `useEffect` inside the `return` of a function component, which is a violation of the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html))